core/roboconf-dm/src/main/java/net/roboconf/dm/internal/api/impl/TargetsMngrImpl.java

```diff
@@ -447,36 +446,33 @@ public String findTargetId( AbstractApplication app, String instancePath ) {
 
 
 	// Finding script resources
+
+
 	@Override
 	public Map<String,byte[]> findScriptResources( String targetId ) throws IOException {
 
-		Map<String,byte[]> result = new HashMap<String,byte[]> ();
+		Map<String,byte[]> result = new HashMap<>( 0 );
 		File targetDir = findTargetDirectory( targetId );
-		String prefix = Utils.removeFileExtension(findTargetFile(targetId, Constants.TARGET_PROPERTIES_FILE_NAME).getName());
-
 		if( targetDir.isDirectory()){
-			List<File> scriptFiles = Utils.listAllFiles(targetDir);
-			for( File scriptFile : scriptFiles) {
-				String name = scriptFile.getName();
-				if( ! CREATED_BY.equals(name)
-						&& name.startsWith(prefix)
-						&& ! name.toLowerCase().endsWith(Constants.FILE_EXT_PROPERTIES)) {
-					ByteArrayOutputStream os = new ByteArrayOutputStream();
-					Utils.copyStream( scriptFile, os );
-					result.put( scriptFile.getName(), os.toByteArray());
-				}
+			result.putAll( Utils.storeDirectoryResourcesAsBytes( targetDir ));
+			result.remove( CREATED_BY );
+
+			// Remove all the root properties files
+			for( File f : Utils.listAllFiles( targetDir )) {
+				if( f.getName().toLowerCase().endsWith( Constants.FILE_EXT_PROPERTIES ))
+					result.remove( f.getName());
 			}
 		}
+
 		return result;
 	}
 
 
 	@Override
-	public Map<String, byte[]> findScriptResources(Application app, Instance scopedInstance) throws IOException {
+	public Map<String, byte[]> findScriptResources( Application app, Instance scopedInstance ) throws IOException {
 
-		Map<String,byte[]> result = new HashMap<String,byte[]> ();
+		Map<String,byte[]> result = new HashMap<> ();
```

Comment:
With Java 7, the implementation can be reduced to `<>` as the generics are already given in the declarated type (Map<String,byte[]>). That's something my IDE did automatically when I saved the document. Since Roboconf requires Java 7 and higher, this change is not a problem. :wink: